### PR TITLE
Convert to use ISM metrics library.

### PR DESCRIPTION
**Pull request by vjkoskela**

### CODE REVIEW

**Review by joeyjackson — APPROVED** [review]
Looks good

File: src/main/java/org/kairosdb/core/http/rest/MetricsResource.java [hunk 1]
```diff
@@ -679,38 +675,23 @@ public Response runQuery(String json, String remoteAddr) throws Exception
...
-				ThreadReporter.addTag("status", "failed");
-			else
-				ThreadReporter.addTag("status", "success");
-
-			//write metrics for query logging
-			long queryTime = System.currentTimeMillis() - ThreadReporter.getReportTime();
-			if (m_logQueries && ((queryTime / 1000) >= m_logQueriesLongerThan))
-			{
-				ThreadReporter.addDataPoint("kairosdb.log.query.remote_address", remoteAddr, m_logQueriesTtl);
-				ThreadReporter.addDataPoint("kairosdb.log.query.json", json, m_logQueriesTtl);
```
[the remark this answers is not in the document]
Comment (vjkoskela) [reply src/main/java/org/kairosdb/core/http/rest/MetricsResource.java:697]: Yes. I'm not sure where KDB was going with the recording of string data; however, I think it's better to log these to disk and we can deal with collecting them separately.

File: src/main/java/org/kairosdb/core/http/rest/MetricsResource.java [hunk 2]
```diff
@@ -860,42 +841,6 @@ private Response executeNameQuery(NameType type, String prefix)
...
-		List<DataPointSet> ret = new ArrayList<>();
-
-		if (count != 0)
-		{
-
-			DataPointSet dpsCount = new DataPointSet(INGEST_COUNT);
-			DataPointSet dpsTime = new DataPointSet(INGEST_TIME);
-
-			dpsCount.addTag("host", hostName);
-			dpsTime.addTag("host", hostName);
```
[the remark this answers is not in the document]
Comment (vjkoskela) [reply src/main/java/org/kairosdb/core/http/rest/MetricsResource.java:877]: Done!

File: src/main/java/org/kairosdb/rollup/RollUpJob.java [hunk 3]
```diff
@@ -120,14 +122,9 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
...
 					finally
 					{
-						ThreadReporter.setReportTime(System.currentTimeMillis());
-						ThreadReporter.clearTags();
-						ThreadReporter.addTag("host", hostName);
-						ThreadReporter.addTag("rollup", rollup.getSaveAs());
-						ThreadReporter.addTag("rollup-task", task.getName());
 						ThreadReporter.addTag("status", success ? "success" : "failure");
-						ThreadReporter.addDataPoint(ROLLUP_TIME, System.currentTimeMillis() - ThreadReporter.getReportTime());
-						ThreadReporter.submitData(longDataPointFactory, publisher);
```
[the remark this answers is not in the document]
Comment (vjkoskela) [reply src/main/java/org/kairosdb/rollup/RollUpJob.java:130]: Removed.

**Pull request merged**